client.go

```diff
@@ -53,3 +58,55 @@ func (c *SortClient) Sort(sortCriteria []SortCriterion, searchCriteria *imap.Sea
 func (c *SortClient) UidSort(sortCriteria []SortCriterion, searchCriteria *imap.SearchCriteria) ([]uint32, error) {
 	return c.sort(true, sortCriteria, searchCriteria)
 }
+
+// NewClient creates a new THREAD client
+func NewThreadClient(c *client.Client) *ThreadClient {
+	return &ThreadClient{c: c}
+}
+
+// SupportThread returns true if the remote server supports the extension.
+func (c *ThreadClient) SupportThread() (bool, error) {
+	for _, capability := range ThreadCapabilities {
+		ok, err := c.c.Support(capability)
+		if err != nil {
+			return false, err
+		} else if ok {
+			return true, err
+		}
+	}
+	return false, nil
+}
+
+func (c *ThreadClient) thread(uid bool, threadAlgorithm string, searchCriteria *imap.SearchCriteria) ([]*Thread, error) {
```

Comment:
We should probably add a type for threading algorithms, with predefined constants.